Code/Framework/AzFramework/AzFramework/Physics/PhysicsScene.h

```diff
@@ -132,6 +132,13 @@ namespace AzPhysics
         //! @return Returns a structure that contains a list of Hits. Depending on flags set in the request, this may only contain 1 result.
         virtual SceneQueryHits QueryScene(SceneHandle sceneHandle, const SceneQueryRequest* request) = 0;
 
+        //! Make a blocking query into the scene.
+        //! @param sceneHandle A handle to the scene to make the scene query with.
+        //! @param request The request to make. Should be one of RayCastRequest || ShapeCastRequest || OverlapRequest
+        //! @param result A structure that contains a list of Hits. Depending on flags set in the request, this may only contain 1 result.
+        //! @return Returns true if there is at least one hit.
+        virtual bool QueryScene(SceneHandle sceneHandle, const SceneQueryRequest* request, SceneQueryHits& result) = 0;
```

Comment:
suggestion: Consider using `AZStd::optional` here as opposed to the `bool` return with out param, it's essentially the same thing and allows RVO to kick in.

Reply:
Do you mean `AZStd::optional<bool>` instead of `bool`? 
Would RVO optimize unnecessary copies of a bool? looks more readable without optional tbh :)

Reply:
No sorry I mean `AZStd::optional<SceneQueryHits>` instead (and no out param)

Reply:
Discussed offline - this API choice was intentional to reduce allocations, sorry for missing this 😅 🙈